File: Modules/RuleEngine/src/hap/ruleengine/parts/property/StringProperty.java
package hap.ruleengine.parts.property;


import hap.ruleengine.parts.Component;

public class StringProperty extends BaseProperty<String>
{
	public StringProperty( String header, String key, String defaultValue, Component c)
	{
		super(header, key, defaultValue, c);
	}

	@Override
	public void setValue( String value )
	{
		myComp.setProperty( myKey, value );
	}

	@Override
	public String getValue()
	{
		return myComp.getProperty( myKey, myDefaultValue );
	}
}

File: Modules/RuleEngine/src/hap/ruleengine/parts/property/StringProperty.kt
package hap.ruleengine.parts.property

import hap.ruleengine.parts.IComponentPropertyAccess
import tornadofx.ValidationMessage
import tornadofx.ValidationSeverity
import tornadofx.observable


class StringProperty(header: String, key: String, defaultValue: String, informationMessage: String, propReader: IComponentPropertyAccess) : BaseProperty<String>(header, key, defaultValue, informationMessage, propReader) {
    override fun validate(value: String?): ValidationMessage {
        var validationMessage = ValidationMessage(null, ValidationSeverity.Success)

        if (value == null) {
            validationMessage = ValidationMessage("Must have a value", ValidationSeverity.Error)
        }

        return validationMessage
    }

    val value = bind { observable(StringProperty::readValue, StringProperty::updateValue) } as javafx.beans.property.StringProperty

    override fun readValue(): String {
        return propReader.getProperty(key, defaultValue)
    }

    override fun updateValue(v: String) {
        propReader.setProperty(key, v)
    }
}
